api/features/workflows/core/models.py

```diff
@@ -169,6 +170,9 @@ def _publish_environment_feature_versions(
                     kwargs={"environment_id": self.environment_id},
                     delay_until=environment_feature_version.live_from,
                 )
+                environment_feature_version_published.send(
+                    EnvironmentFeatureVersion, instance=environment_feature_version
+                )
```

Comment:
I can't see from this review where `environment_feature_version_published` is actually coming from.

Reply:
It's a signal defined [here](https://github.com/Flagsmith/flagsmith/blob/main/api/features/versioning/signals.py) and consumed [here](https://github.com/Flagsmith/flagsmith/blob/main/api/features/versioning/receivers.py#L42-L64).

I kind of hate using signals because it can be a bit hard to track the execution stack, but I seem to remember having no choice in this case.